tests/foreman/api/test_subscription.py

```diff
@@ -240,6 +243,53 @@ def test_positive_subscription_status_disabled(self):
         host_content = entities.Host(id=host_id).read_raw().content
         assert 'Simple Content Access' in str(host_content)
 
+    @pytest.mark.tier2
+    @pytest.mark.usefixtures('golden_ticket_host_setup')
+    @pytest.mark.usefixtures('rhel77_contenthost_class')
+    def test_sca_end_to_end(self):
+        """Perform end to end testing for Simple Content Access Mode
+
+        :id: c6c4b68c-a506-46c9-bd1d-22e4c1926ef8
+
+        :expectedresults: All end to end tests pass and clients have access
+        to repos without needing to add subscriptions
+
+        :CaseImportance: Medium
+        """
+        self.content_host.install_katello_ca()
+        self.content_host.register_contenthost(self.org_setup.label, self.ak_setup.name)
+        assert self.content_host.subscribed
+        # Check to see if Organization is in SCA Mode
+        assert entities.Organization(id=self.org_setup.id).read().simple_content_access is True
+        # Verify that you cannot attach a subscription to an activation key in SCA Mode
+        subscription = entities.Subscription(organization=self.org_setup).search(
+            query={'search': f'name="{DEFAULT_SUBSCRIPTION_NAME}"'}
+        )[0]
+        with pytest.raises(HTTPError) as ak_context:
+            self.ak_setup.add_subscriptions(
+                data={'quantity': 1, 'subscription_id': subscription.id}
+            )
+        assert "Simple Content Access" in ak_context.value.response.text
+        # Verify that you cannot attach a subscription to an Host in SCA Mode
+        host_id = entities.Host().search(query={'search': self.content_host.hostname})[0].id
+        with pytest.raises(HTTPError) as host_context:
+            entities.HostSubscription(host=host_id).add_subscriptions(
+                data={'subscriptions': [{'id': subscription.id, 'quantity': 1}]}
+            )
+        assert "Simple Content Access" in host_context.value.response.text
+        # Create a content view with repos and check to see that the client has access
+        content_view = entities.ContentView(organization=self.org_setup).create()
+        content_view.repository = [self.rh_repo_setup, self.custom_repo_setup]
+        content_view.update(['repository'])
+        content_view.publish()
+        host = entities.Host().search(query={'search': self.content_host.hostname})[0]
+        host.content_facet_attributes = {'content_view_id': content_view.id}
+        host.update(['content_facet_attributes'])
+        self.content_host.run('subscription-manager repos --enable *')
+        repos = self.content_host.run('subscription-manager refresh && yum repolist')
+        assert content_view.repository[1].name in repos.stdout
```

Comment:
I would also add a cv length check here so ensure that the correct repo count shows up since you're comparing the 2 at the end.

And regarding a better way to do the assert, you'll have to do some manipulation on how `repos` stores it's content.  It's unfortunate that it's all one string of text and can't be broken apart.  Maybe there's a different command that just returns the repo names and nothing else?